Add the preference query parameter options (#4099)
opensearch-project/documentation-website: _api-reference/search.md

lenient | Boolean | Specifies whether OpenSearch should accept requests if queries have format errors (for example, querying a text field for an integer). Default is false.
max_concurrent_shard_requests | Integer | How many concurrent shard requests this request should execute on each node. Default is 5.
pre_filter_shard_size | Integer | A prefilter size threshold that triggers a prefilter operation if the request exceeds the threshold. Default is 128 shards.

> preference | String | Specifies on which shard or node OpenSearch should perform the search. For valid values, see [The `preference` query parameter](#the-preference-query-parameter). 

Reviewer: ```suggestion
preference | String | Specifies the shards or nodes on which OpenSearch should perform the search. For valid values, see [The `preference` query parameter](#the-preference-query-parameter). 
```

typed_keys | Boolean | Whether returned aggregations and suggested terms should include their types in the response. Default is true.
version | Boolean | Whether to include the document version as a match.

### The `preference` query parameter

> The `preference` query parameter specifies on which shard or node OpenSearch should perform the search. Valid values are:

Reviewer: ```suggestion
The `preference` query parameter specifies the shards or nodes on which OpenSearch should perform the search. Valid values are:
```

> The `preference` query parameter specifies the shards or nodes on which OpenSearch should perform the search. Valid values are:

Reviewer 2: ```suggestion
The `preference` query parameter specifies the shards or nodes on which OpenSearch should perform the search. The following are valid values:
```

> The `preference` query parameter specifies the shards or nodes on which OpenSearch should perform the search. Valid values are:

Reviewer 2: Where possible, the text preceding the colon should be a complete clause.

- `_primary`: Perform the search only on primary shards.
- `_replica`: Perform the search only on replica shards.

> - `_primary_first`: Perform the search on primary shards, but fail over to other available shards if primary shards are not available.

Reviewer 2: ```suggestion
- `_primary_first`: Perform the search on primary shards but fail over to other available shards if primary shards are not available.
```

> - `_replica_first`: Perform the search on replica shards, but fail over to other available shards if replica shards are not available.

Reviewer 2: ```suggestion
- `_replica_first`: Perform the search on replica shards but fail over to other available shards if replica shards are not available.
```

> - `_local`: If possible, perform search on the local node's shards.

Reviewer 3: ```suggestion
- `_local`: If possible, perform the search on the local node's shards.
```

> - `_prefer_nodes:<node-id-1>,<node-id-2>`: If possible, perform search on the specified nodes. Use a comma-separated list to specify multiple nodes.

Reviewer 3: ```suggestion
- `_prefer_nodes:<node-id-1>,<node-id-2>`: If possible, perform the search on the specified nodes. Use a comma-separated list to specify multiple nodes.
```

- `_shards:<shard-id-1>,<shard-id-2>`: Perform the search only on the specified shards. Use a comma-separated list to specify multiple shards. When combined with other preferences, the `_shards` preference must be listed first. For example, `_shards:1,2|_replica`.
- `_only_nodes:<node-id-1>,<node-id-2>`: Perform the search only on the specified nodes. Use a comma-separated list to specify multiple nodes.

> - `<string>`: Specifies a custom string to use for search. The string cannot start with an underscore character (`_`). Searches with the same custom string are routed to the same shards.

Reviewer 2: ```suggestion
- `<string>`: Specifies a custom string to use for the search. The string cannot start with an underscore character (`_`). Searches with the same custom string are routed to the same shards.
```
